src/modules/mavlink/streams/STATUSTEXT.hpp

```diff
@@ -61,18 +61,17 @@ class MavlinkStreamStatustext : public MavlinkStream
 
 	bool send() override
 	{
-		if (_mavlink->is_connected() && (_mavlink->get_free_tx_buf() >= get_size())) {
+		const unsigned last_generation = _mavlink_log_sub.get_last_generation();
 
-			const unsigned last_generation = _mavlink_log_sub.get_last_generation();
+		mavlink_log_s mavlink_log;
 
-			mavlink_log_s mavlink_log;
-
-			if (_mavlink_log_sub.update(&mavlink_log)) {
-				if (_mavlink_log_sub.get_last_generation() != (last_generation + 1)) {
-					_missed_msg_count++;
-					PX4_ERR("channel %d has missed %d mavlink log messages", _mavlink->get_channel(), _missed_msg_count);
-				}
+		if (_mavlink_log_sub.update(&mavlink_log)) {
+			if (_mavlink_log_sub.get_last_generation() != (last_generation + 1)) {
+				_missed_msg_count++;
+				PX4_ERR("channel %d has missed %d mavlink log messages", _mavlink->get_channel(), _missed_msg_count);
+			}
 
+			if (_mavlink->is_connected() && (_mavlink->get_free_tx_buf() >= get_size())) {
```

Comment:
On second thought, I think this kind of misses the point. The original intent was to try and get a sense of which messages weren't ever making it out. Some of this can be from the mavlink module not keeping up, etc, but the larger portion of dropped messages in real usage comes from the TX buffer check.

How about we explicitly handle the case where it's first able to transmit? Then the missed mavlink message count/error would only apply from that point forward?

Reply:
@dagar please change it as you think it makes sense. All I did was fix the issue where bogus statustexts were arriving.